app/src/main/java/com/tananaev/logcat/Line.java

```diff
@@ -15,30 +15,56 @@
  */
 package com.tananaev.logcat;
 
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class Line {
 
-    private static Pattern linePattern = Pattern.compile("\\d\\d-\\d\\d \\d\\d:\\d\\d:\\d\\d\\.\\d\\d\\d (\\w)/.*");
+    private static Pattern linePattern = Pattern.compile("\\d\\d-\\d\\d \\d\\d:\\d\\d:\\d\\d\\.\\d\\d\\d (\\w)/(\\w+).*");
 
-    private char level = 'D';
-    private String content;
+    private final char level;
+    @NonNull
+    private final String lowerTag;
+    @NonNull
+    private final String content;
+    @Nullable
+    private String lowerContent;
```

Comment:
Do we really need to have `lowerTag` and `lowerContent`? It basically means that we store double the amount of data required. I think we should remove it and just use original strings.